core/org.openjdk.jmc.common/src/main/java/org/openjdk/jmc/common/IMCFrame.java

```diff
@@ -49,45 +51,82 @@ public interface IMCFrame {
 	/**
 	 * Frame compilation types.
 	 */
-	public enum Type {
-	/**
-	 * The frame was executed as native code compiled by the Java JIT compiler.
-	 */
-	JIT_COMPILED,
-	/**
-	 * The frame was executed as interpreted Java byte code.
-	 */
-	INTERPRETED,
-	/**
-	 * The frame was executed as code that was inlined by the Java JIT compiler.
-	 */
-	INLINED,
-	/**
-	 * The frame was executed as native code, most probably a C function.
-	 */
-	NATIVE,
-	/**
-	 * The frame was executed as native code compiled from C++.
-	 */
-	CPP,
-	/**
-	 * The frame was executed as kernel native code.
-	 */
-	KERNEL,
-	/**
-	 * The frame compilation type is unknown.
-	 */
-	UNKNOWN;
+	final class Type {
+		private static final String MSG_PREFIX = "IMCFrame_Type_";
+
+		private final String id;
+		private final String name;
+		private final boolean isUnknown;
 
-		private String name;
+		public Type(String id) {
+			this.id = id.toUpperCase();
 
-		private Type() {
-			name = Messages.getString("IMCFrame_Type_" + toString()); //$NON-NLS-1$
+			String key = MSG_PREFIX + this.id;
+			if (Messages.hasString(key)) {
+				this.name = Messages.getString(key);
+				this.isUnknown = false;
+			} else {
+				this.name = this.id;
+				this.isUnknown = true;
+			}
 		}
 
+		/**
+		 * The frame was executed as native code compiled by the Java JIT compiler.
+		 */
+		public static final Type JIT_COMPILED = new Type("JIT_COMPILED"); //$NON-NLS-1$
+		/**
```

Comment:
I would like if the static type definitions could move up, directly behind `final class Type {` and were separated by a blank line.